### ISSUE TITLE
fix: plugin anchor-navigation-ex dependency issue with colors@1.4.1

### ISSUE THREAD

**Issue by onejar99** [onejar99/gitbook-build-publish-action#3]
[situation]
user ticket: https://github.com/onejar99/demo-build-and-publish-gitbook/issues/2

When `gitbook build`, loading plugin `anchor-navigation-ex` will cause a weird crash and stick the process.
Reviewed dependency tree and discovered the `colors@1.4.1` is the root cause. `colors@1.4.0` was fine.

error:
```
root@659a36a58fc5:/gitbook# gitbook install
info: installing 5 plugins using npm@3.9.2
info:
info: installing plugin "anchor-navigation-ex"
info: install plugin "anchor-navigation-ex" (*) from NPM with version 1.0.14
/gitbook
`-- gitbook-plugin-anchor-navigation-ex@1.0.14
  +-- cheerio@0.22.0
  | +-- css-select@1.2.0
  | | +-- boolbase@1.0.0
  | | +-- css-what@2.1.3
  | | +-- domutils@1.5.1
  | | `-- nth-check@1.0.2
  | +-- dom-serializer@0.1.1
  | | `-- domelementtype@1.3.1
  | +-- entities@1.1.2
  | +-- htmlparser2@3.10.1
  | | +-- domhandler@2.4.2
  | | +-- inherits@2.0.4
  | | `-- readable-stream@3.6.0
  | |   +-- string_decoder@1.3.0
  | |   | `-- safe-buffer@5.2.1
  | |   `-- util-deprecate@1.0.2
  | +-- lodash.assignin@4.2.0
  | +-- lodash.bind@4.2.1
  | +-- lodash.defaults@4.2.0
  | +-- lodash.filter@4.6.0
  | +-- lodash.flatten@4.4.0
  | +-- lodash.foreach@4.5.0
  | +-- lodash.map@4.6.0
  | +-- lodash.merge@4.6.2
  | +-- lodash.pick@4.4.0
  | +-- lodash.reduce@4.6.0
  | +-- lodash.reject@4.6.0
  | `-- lodash.some@4.6.0
  +-- colors@1.4.1
  `-- github-slugid@1.0.1

info: >> plugin "anchor-navigation-ex" installed with success
info:
info: installing plugin "edit-link"
runTopLevelLifecycles ->  / |##############################################################################################---|
/gitbook
+-- gitbook-plugin-anchor-navigation-ex@1.0.14
`-- gitbook-plugin-edit-link@2.0.2

info: >> plugin "edit-link" installed with success
info:
info: installing plugin "copy-code-button"
runTopLevelLifecycles ->  \ |##############################################################################################---|
/gitbook
+-- gitbook-plugin-anchor-navigation-ex@1.0.14
+-- gitbook-plugin-copy-code-button@0.0.2
`-- gitbook-plugin-edit-link@2.0.2

info: >> plugin "copy-code-button" installed with success
info:
info: installing plugin "theme-comscore"
runTopLevelLifecycles ->  \ |##############################################################################################---|
/gitbook
+-- gitbook-plugin-anchor-navigation-ex@1.0.14
+-- gitbook-plugin-copy-code-button@0.0.2
+-- gitbook-plugin-edit-link@2.0.2
`-- gitbook-plugin-theme-comscore@0.0.3

info: >> plugin "theme-comscore" installed with success
info:
info: installing plugin "ga"
runTopLevelLifecycles ->  / |##############################################################################################---|
/gitbook
+-- gitbook-plugin-anchor-navigation-ex@1.0.14
+-- gitbook-plugin-copy-code-button@0.0.2
+-- gitbook-plugin-edit-link@2.0.2
+-- gitbook-plugin-ga@1.0.1
`-- gitbook-plugin-theme-comscore@0.0.3

info: >> plugin "ga" installed with success
root@659a36a58fc5:/gitbook# gitbook build
info: 12 plugins are installed
info: 11 explicitly listed
info: loading plugin "anchor-navigation-ex"... LIBERTY LIBERTY LIBERTY
LIBERTY LIBERTY LIBERTY
LIBERTY LIBERTY LIBERTY
                                   !             H|H|H|H|H           H__________________________________             H|§|§|§|H           H|* * * * * *|---------------------|             H|§|∞|§|H           H| * * * * * |---------------------|             H|§|§|§|H           H|* * * * * *|---------------------|             H|H|H|H|H           H| * * * * * |---------------------|             H|H|H|H|H           H|---------------------------------|          ===============        H|---------------------------------|            /| _   _ |          H|---------------------------------|            (| O   O |)          H|---------------------------------|            /|   U   |          H-----------------------------------             |  =/  |           H              _..._/            H              _|I/|_            H      _______/| H |/_______    H     /           / /          H    |          | | /         |  H    |          ||o||          |  H    |    |     ||o||     |    |  H    |    |     ||o||     |    |  H   Carl Pilcher
t̛͌̐̍̅ͣ͆̿͊̌͠ ҉͎̟͍̠̤͍̜͇̳̼̳̤̜̟̙̯̻̝ę̛͙͓͎̘̯̳̯̘̳͊̒̓̽̃̌͜͝s̴̡̳̞͚̹̃͒ͣ͂̏ͩͨͨͥ̒t̷̐̎̆̑ͬ̇̄̓̈̄̿ͩ̄̂̐̃͂̌ȋ̡̟̫̳͓͉̟̇ͨͣͮ̑̾̎̔̚̚͞n̴̨̧̠̭͕̉̾ͮͮͫ̓ͮ̑̈̂̾͜͡g̶̢̘̫̥̪̰̝̠͓̥̮ͯ͒̏͂͘͟ͅ ̷̵̼̻͍͉̜̘͖͓̓̒̇̍ͤͧ̈͜͞t̷̵̵͚̻͍̠̺̹̗̥̼̓͌ͮ͐͑̌ͅe̢͖̼͉͔̞̰͚̤͗ͬ̐̃ͫ͆͂̎̓̕s̶̪̖͙̱̬͖̖̒ͤ̽̽ͭ̑̆ͥ̀ͅͅt͌ͭͥ͌̑ͭͨ͋ͦ̃ͮͯ̊̀ͤ̈ͣ͊́ĭ̧̥͋͐͊̈͛ͯ̌̇̀̿̆̓̂̉̉̀n̶̡͈͉͕͉̟ͫͬͥ̂̀ͩ̿̉̾́́͢g̴̡̛̯͕̺̫̿̀ͫ͒ͩ͗ͫ̏̃ͧͪ́ ̵̢̝̦͕̯̗̠̗̩͖͎͐͂̂̀̋͛͊t̶̸̢̻̟̱̪̪͍̔̐̌́ͦ̃̌͑̾͜e̵̷̓̓ͭͥ̄ͨ͒̋̓ͪ̈̔͊̊́̈̕ṡ̓ͮ͌̆ͭ̅̀̀ͯ̔̓̎ͩͯͣ̆̽ͬţ̸̨̛͕̲̙̲̒̈͊ͯ̏̊͊ͭ͋͌̾i̡͉̬̹̞ͪ̑̈ͧ͒ͩ̈̒ͮ̌͗̄́͘nͧ̓̄̿͛̊ͧ̓͒͛̽̄ͦ̋ͯ̔̏͑͛gͣ̑ͥ̎̇̇̓̌̐̿ͣͪͫ͛ͨ͊̂ͬ̚ ̵̸̶̧ͮ̓̔̋͛͐̑ͦ̀̔̿̌ͨ ҉̫̖̟͉͍̦̳̩͖̫̥̲̩̭̤͉̖ͅt͎̣͔͓͇̤̩̫̘ͥ̉̌ͬ͂̈ͫ̀̚͘ȩ̢̛͕̣̣̦͚̬͎͆͒̈ͥͥͨͭ͂ͯs̙͇̺̯̈ͧ́ͦ́ͬ̐ͦ͂̆ͬ͂̈͘͟t̵̴̮̮͉͇̣̉ͤ̔ͥ̇̾̒̿̈̋̚ͅi̴͚͚̲̝̥̹̞̤̩̻͖ͣ̑͋̈ͣ͡ͅn̽̿ͦ̄́̆ͣ̈͐ͭͪ̊ͪ ҉̴̸͢ ҉͚͙͈̜̳̘̟̥̬̞̰̻̟̬̜͓̝͙ģ̷̢̜͙̞̖͖̤̗̯̖͕̈̃̈ͮ͘͢ ̨̬̼̬̣̋̐̒̈ͮ͗̉̓̋ͥ̏̃͑͘t̤͖̼͍̙̣̲̯̦̿̿̈̉͆͊̓́͘͝ȅ̵̢̢̡͎̱̻̈͌͐̑͌̋̎̆̔ͪ͛s̠͇̳̹̙̭͂͗̏̈̀ͥͦ̂ͥ̚̚͠͞t̴̛̬̫̥͕͇̲̥̍̇̂̂̔̂̐́͘͞i̽ͦ̎ͤͫͥ̽́̊͛̓̌ ҉͙͓͚͎̙̰̥̼̰͍̤͔̼͈̫̦̀ͅn̡̨̞͙̟͖͙͓̞̩̅ͯ͆ͨͬͥ͢͝͡g̢̢̊͌͆͆ͨ̅͛͜ ҉̙͖̤͚̗̯̬͓̩̣̺͍̺͉̰̜̖̙ ̸̨̜̣͖̿͐͒͗̊͒ͥ͑̅̏ͫ͗͢͡ẗ̡͂͑̋̂̂̌͂ͫ̀ͯ̃ͣͦ͗̎̌́ē̠ͭ͆̓̇͆̎̇̎͗̂̆ͣ̄̇̄͘͝s̴̢̩̼͎̭̹͚͒̑ͭͨ̑̿̈ͪ͢͡ͅt̨̟̮̖̲̳̰̯̑̒̑͌̊ͧ̈́̕͝ͅi̸ͨ̑̽̅́̏̇̎͒̒̃͂ͫͦ͌͝͞͡ņ̛̯̠̓͒̅̈ͫ̽ͣͮͤ̆̃̔̄̔͘g̶̶̡͚͙̱͉̾̏̏͗ͪͮ͌̿̈̋̀ͅ ̴̘͉ͨͣͬ̋̂̓̿̑ͦ͋̏ͪ͐̒́͜t̴̢ͥ͒͑͂͒͛͂ͭͥ̃͋ͮͭ̏͑̓͢e̠̳̟̣̝̰̼̹̲̜̰͍͕̽̔̿͊͜͠s̢̝̩̗̭̗̩̦̯̟̝ͩ̾̐̆͌̈́͢t̸̴̲͕̱̦͚͎̠̬ͩ͐̏̍ͣ̾̂̂͆ĭ̵̻̹͈̫͎͍̩̗̠̹̰̋ͦ̆͠ͅͅn̴̡̡͎͕͎̤̣̠̯̘ͤ͂ͦ̉͆ͪ̅ͣģͭ̏̔́̌̽̇̾̿ ҉̷̝̥̖̙̬̳̗̺͕̪̯̗͘Arraẏ̡̛̗̱̼͕̤ͮ̈ͦ͛͗͗ͪ͌ͮ̽ͧfunction(symbol) {
      var $a, self = this, $iter = TMP_5.$$p, block = $iter || nil, $splat_index = nil;

      var array_size = arguments.length - 1;
      if(array_size < 0) array_size = 0;
      var args = new Array(array_size);
      for($splat_index = 0; $splat_index < array_size; $splat_index++) {
        args[$splat_index] = arguments[$splat_index + 1];
      }
      TMP_5.$$p = null;
      return $scope.get('Kernel').$raise($scope.get('NoMethodError').$new((function() {if ((($a = self.$inspect && !self.$inspect.$$stub) !== nil && (!$a.$$is_boolean || $a == true))) {
        return "undefined method `" + (symbol) + "' for " + (self.$inspect()) + ":" + (self.$$class)
        } else {
        return "undefined method `" + (symbol) + "' for " + (self.$$class)
      }; return nil; })(), symbol));
    }̢͙̟̦͖͌̇̐͂̇ͣ͊ͪ͛̇̌͟͢͝function() {

```

**Comment by onejar99** [comment]
[solution]
pre-install `colors@1.4.0` before `gitbook install`.
Released with v1.0.2 and verified.

**Comment by onejar99** [comment]
note the available dependency tree for reference:
```
root@9f7d802fd088:/gitbook# npm list
/gitbook
+-- gitbook-plugin-anchor-navigation-ex@1.0.14
| +-- cheerio@0.22.0
| | +-- css-select@1.2.0
| | | +-- boolbase@1.0.0
| | | +-- css-what@2.1.3
| | | +-- domutils@1.5.1
| | | | +-- dom-serializer@0.1.1 deduped
| | | | `-- domelementtype@1.3.1 deduped
| | | `-- nth-check@1.0.2
| | |   `-- boolbase@1.0.0 deduped
| | +-- dom-serializer@0.1.1
| | | +-- domelementtype@1.3.1
| | | `-- entities@1.1.2 deduped
| | +-- entities@1.1.2
| | +-- htmlparser2@3.10.1
| | | +-- domelementtype@1.3.1 deduped
| | | +-- domhandler@2.4.2
| | | | `-- domelementtype@1.3.1 deduped
| | | +-- domutils@1.5.1 deduped
| | | +-- entities@1.1.2 deduped
| | | +-- inherits@2.0.4
| | | `-- readable-stream@3.6.0
| | |   +-- inherits@2.0.4 deduped
| | |   +-- string_decoder@1.3.0
| | |   | `-- safe-buffer@5.2.1
| | |   `-- util-deprecate@1.0.2
| | +-- lodash.assignin@4.2.0
| | +-- lodash.bind@4.2.1
| | +-- lodash.defaults@4.2.0
| | +-- lodash.filter@4.6.0
| | +-- lodash.flatten@4.4.0
| | +-- lodash.foreach@4.5.0
| | +-- lodash.map@4.6.0
| | +-- lodash.merge@4.6.2
| | +-- lodash.pick@4.4.0
| | +-- lodash.reduce@4.6.0
| | +-- lodash.reject@4.6.0
| | `-- lodash.some@4.6.0
| +-- colors@1.4.0
| `-- github-slugid@1.0.1
+-- gitbook-plugin-copy-code-button@0.0.2
+-- gitbook-plugin-edit-link@2.0.2
+-- gitbook-plugin-ga@1.0.1
`-- gitbook-plugin-theme-comscore@0.0.3
```
